schema.cc

```diff
@@ -432,6 +432,21 @@ schema::schema(const schema& o)
     init_view_info_from_other(o);
 }
 
+schema::schema(const schema& s, reversed_tag)
+    : _raw(s._raw)
+    , _offsets(s._offsets)
+{
+    _raw._version = utils::UUID(s._raw._version.get_most_significant_bits(), ~s._raw._version.get_least_significant_bits());
+    for (auto& col : _raw._columns) {
+        if (col.kind == column_kind::clustering_key) {
+            col.type = reversed(col.type);
+        }
+    }
+
+    rebuild();
+    init_view_info_from_other(s);
```

Comment:
It's sad we need to repeat those obscure steps. It's very possibly someone will add another weird step in the future and we'll forget that this function needs update as well. Couldn't we used schema's copy constructor to copy the original schema entirely, and then go back and modify only the parts we want to modify (_raw.version and _raw.columns)?

Reply:
Yes, I've spent hours debugging a test only to find out I was lacking these obscure steps in my new constructor. We can't use the copy constructor because we have to inject our step *after* copying the raw schema but *before* calling `rebuild()`.

Reply:
Maybe we should have a private copy constructor shared by all copy constructors which allows you to inject a transformation step.